src/query/QueryManager.js

```diff
@@ -0,0 +1,63 @@
+import ObjectSelector from './ObjectSelector';
+import SwisstopoIdentify from './SwisstopoIdentify';
+import ScreenSpaceEventType from 'cesium/Source/Core/ScreenSpaceEventType';
+import i18next from 'i18next';
+
+
+export default class QueryManager {
+  constructor(viewer) {
+    this.objectSelector = new ObjectSelector(viewer);
+    this.swisstopoIndentify = new SwisstopoIdentify();
+    this.scene = viewer.scene;
+    viewer.screenSpaceEventHandler.setInputAction(click => this.onclick(click), ScreenSpaceEventType.LEFT_CLICK);
+  }
+
+  set activeLayers(names) {
+    this.searchableLayers = names;
+  }
+
+  async querySwisstopo(pickedPosition, layers) {
+    const lang = i18next.language;
+    const identifyData = await this.swisstopoIndentify.identify(pickedPosition, layers, lang);
+    if (identifyData) {
+      const d = identifyData;
+      const {layerBodId, featureId} = identifyData;
+      let popupContent = await this.swisstopoIndentify.getPopupForFeature(layerBodId, featureId, lang);
+      if (popupContent) {
+        popupContent = popupContent.replace(/cell-left/g, 'key')
+        .replace(/<td>/g, '<td class="value">')
+        .replace(/<table>/g, '<table class="ui compact small very basic table">');
+      }
+      const onshow = () => {
+        console.log('showing', d.geometry);
+      };
+      const onhide = () => {
+        console.log('hiding', d.geometry);
+      };
+      return {
+        popupContent,
+        onhide,
+        onshow
+      };
+    }
+  }
+
+  async onclick(click) {
```

Comment:
There are some issues with a click:

1. When clicking out of map it returns the error: https://i.imgur.com/wtJvdN6.jpeg
2. When drawing popup appears: https://i.imgur.com/dY8oiTW.jpeg

Reply:
I fixed these, thanks.

Reply:
You disabled query when tools are open and now AOI cant be selected when AOI section is open